include/class.canned.php

```diff
@@ -229,11 +229,25 @@ static function getIdByTitle($title) {
     }
 
     static function getCannedResponses($deptId=0, $explicit=false) {
+        global $thisstaff;
+
         $canned = static::objects()
             ->filter(array('isenabled' => true))
             ->order_by('title')
             ->values_flat('canned_id', 'title');
 
+        if ($thisstaff) {
+            $departments = array();
+            //get responses for all depts
+            $departments[] = 0;
+            //get the responses for the agent's departments
+            $staffDepts = $thisstaff->getDepartmentNames(true);
+            foreach ($staffDepts as $k => $v)
```

Comment:
Instead of foreach look I would suggest then add the zero key thereafter.
```php
$departments = array_keys($thisstaff->getDepartmentNames(true));
```